kubelet/datadog_checks/kubelet/kubelet.py

```diff
@@ -223,6 +233,44 @@ def _create_kubelet_prometheus_instance(self, instance):
         )
         return kubelet_instance
 
+    def _create_pod_tags_by_pvc(self, pods):
+        """
+        Return a map, e.g.
+            {
+                "<persistentvolumeclaim>|<kube_namespace>": [<list_of_pod_tags>],
+                "<persistentvolumeclaim1>|<kube_namespace1>": [<list_of_pod_tags1>],
+            }
+        that can be used to add pod tags to associated volume metrics
+        """
+        pod_tags_by_pvc = {}
+        for pod in pods['items']:
+            # get persistentvolumeclaim
+            volumes = pod.get('spec', {}).get('volumes')
+            if not volumes:
+                continue
+            for v in volumes:
+                pvc_name = v.get('persistentVolumeClaim', {}).get('claimName')
+                if pvc_name:
+                    break
+            if not pvc_name:
+                continue
+
+            # get kubernetes namespace of pvc
+            kube_ns = pod.get('metadata', {}).get('namespace')
+            if not kube_ns:
+                continue
+
+            # get pod tags from tagger
+            pod_id = pod.get('metadata', {}).get('uid')
+            if not pod_id:
+                self.log.debug('skipping pod with no uid')
+                continue
+            tags = tagger.tag('kubernetes_pod_uid://%s' % pod_id, tagger.LOW) or None
+            if not tags:
+                continue
+            pod_tags_by_pvc['{}|{}'.format(pvc_name, kube_ns)] = tags
```

Comment:
nit: we usually reference k8s objects with the format: `<namespace>/<name>`

Reply:
what if the PVC is attached to multiple pods? we're overwriting the entry in the map so we'll only get the tags of the last visited pod, let's `extend` the list of tags if the element already exists in the map and add unit test for this case